### PR TITLE
Add files via upload

### PR DESCRIPTION
**Pull request by ahmadbougham**
Dear Nicholas,
 I have tried to improve the MLP class by using the F-adjoint formulation which is introduced in my recent arxiv preprint :  "https://arxiv.org/abs/2304.13820" (see also my introduction about this formulation in  "https://github.com/ahmadbougham/F-adjoint"). It seems that the accuracy is of order 94%.

Any feedback you gave me would be highly appreciated.

Sincerely,
Ahmed

### CODE REVIEW

**Comment by nicholasbaard** [comment]
Hi Ahmed! Thank you so much for wanting to collaborate and improve the Repo!

There are a few things I would like you to consider before I merge this into the main branch:

- I would like to keep the old backpropagation function as well as your F-adjoint implementation. Please could you add the backprop back and make the MLP class generalisable so that a user can decide whether they wish to use backprop or F star. I think this would be good for viewing experiments with the two methods.
- The user should be able to select backpropagation/F-adjoint as an argument in the argparser.
- I would like you to please use the naming conventions that were originally used. This means "W{i}" for weight matrices, "A{i}" for activation layers, "b{i}" for bias nodes, "delta{i}" and "grad{i}" for deltas and gradient matrices. This is so that the naming matches the notation in the README.
- I would like you to add the F-adjoint information to the readme, explaining the benefits and the improvements compared to backpropagation as well as links to your paper.

Once again, thank you for your contribution!

**Comment by ahmadbougham** [comment]
Dear Nic, thank you for your rapid response. Let me inform you that:

1- I am a Dr in Mathematics and I am interesting, recently, to study the maths behind Artificial Neural Networks.

2- I have just only the minimum competences on implementation/coding programs.

So, I have trying to improve my experience via eventual collaborations. By the way, I have made my best to provide you the requested tasks and I, kindly ask you to help me to go further in this project. Think you very much in advance.
